src/lib/yaraparse.py

```diff
@@ -476,11 +479,37 @@ def of_expr(self, condition: OfExpression) -> Optional[UrsaExpression]:
             return None
 
     def gt_expr(self, condition: GtExpression) -> Optional[UrsaExpression]:
-        left = self.traverse(condition.left_operand)
-        right = self.traverse(condition.right_operand)
-        return left or right
+        """ From `#a > 3` we can deduce, that #a must occur at least once """
+        return self.traverse(condition.left_operand)
+
+    def lt_expr(self, condition: LtExpression) -> Optional[UrsaExpression]:
+        """ From `3 < #a` we can deduce, that #a must occur at least once """
+        return self.traverse(condition.right_operand)
+
+    def ge_expr(self, condition: GeExpression) -> Optional[UrsaExpression]:
+        """ From `#a >= y` we can deduce, that #a must occur at least once if y>0 """
+        if isinstance(condition.right_operand, IntLiteralExpression):
+            if condition.right_operand.value > 0:
+                return self.traverse(condition.left_operand)
+
+    def le_expr(self, condition: LeExpression) -> Optional[UrsaExpression]:
+        """ From `y <= #a` we can deduce, that #a must occur at least once if y>0 """
+        if isinstance(condition.left_operand, IntLiteralExpression):
+            if condition.left_operand.value > 0:
+                return self.traverse(condition.right_operand)
 
     def eq_expr(self, condition: EqExpression) -> Optional[UrsaExpression]:
+        """ From `x = y` we can deduce, that x and y must both occur - unless:
+            - x = 0, or y = 0
+            - both x and y are count expressions (both can be zero)
+        """
+        if isinstance(condition.left_operand, IntLiteralExpression):
+            if isinstance(condition.right_operand, IntLiteralExpression):
+                if condition.left_operand == condition.right_operand == 0:
```

Comment:
shouldn't this be checked separately for both sides? i.e. 
```python
if isinstance(condition.left_operand, IntLiteralExpression):
    if condition.left_operand == 0:
        return None
if isinstance(condition.right_operand, IntLiteralExpression):
    if condition.right_operand == 0:
        return None
```
Right now it only check for `0 = 0`  which seems kinda bizarre :thinking:

Reply:
I've simplified this condition a lot. I wasn't confident enough that this is correct (what if yara adds another type of expression with another semantics?).